### PR TITLE
fix: stop informer not more valid and handle similar contexts

### LINKED ISSUE #6830: Cluster unreachable even though i can connect
### Bug description

I'm working on a demo using kind and it happened multiple times that desktop shows that the cluster is unreachable even though i can connect to it. If i stop and restart desktop it works again.

One thing to replicate is when you create a new kind cluster. After the creation it tells it is not reachable.

### Operating system

Windows 11

### Installation Method

None

### Version

next (development version)

### Steps to reproduce

_No response_

### Relevant log output

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by lstocchi**
### What does this PR do?

This PR fixes some weird behavior with the informers.
So far, we were just using the context name to detect if a context was deleted, updated, added but it may happen that a context only differs for some internal value (e.g the cluster server url). E.g create podman machine, create kind cluster, delete podman machine, the kind cluster is deleted but the kubeconfig remains dirty, so if you create a new kind cluster, the kubeconfig is almost the same expect the server url and the informer are not updated

Another thing is that we restart an informer if it fails at starting. This way, it may happen that you end up having multiple informers even though you only have one cluster. This PR adds a check to prevent a context that do not belong to the kubeconfig to be restarted again.

E.g. As before you start with a dirty kubeconfig. 
Podman desktop starts an informer which will fail as the cluster is not reachable and it will try to restart it again and again after a timeout 

https://github.com/containers/podman-desktop/blob/654b227832b93b27631507ca7cb9e09e603cf3b3/packages/main/src/plugin/kubernetes-context-state.ts#L790
 
In the meantime you create a new kind cluster.
Desktop will try to stop the old informer by calling the kubeclient library stop method (without any success as there is no active informer, we restart it on desktop side), and create a new informer pointing to the new cluster. 
You end up having a working informer and a failing one that keep trying at starting infinitely.

### Screenshot / video of UI

N/A

### What issues does this PR fix or reference?

it resolves #6830 

### How to test this PR?

1. create a podman machine 
2. create a kind cluster
3. delete the podman machine
4. your kubeconfig should still have the entry regarding the kind cluster
5. recreate a podmna machine + kind cluster having the same name as before
6. if you go to the deployment/service/ingress it should be connected to the newly created cluster

- [x] Tests are covering the bug fix or the new feature

### CODE REVIEW

File: packages/main/src/plugin/kubernetes-context-state.spec.ts [hunk 1]
```diff
@@ -305,8 +306,10 @@ describe('update', async () => {
     });
 
     apiSenderSendMock.mockReset();
+    console.log('DA QUI');
```
Comment (axel7083) [inline packages/main/src/plugin/kubernetes-context-state.spec.ts]:
```suggestion
```

File: packages/main/src/plugin/kubernetes-context-state.spec.ts [hunk 2]
```diff
@@ -330,6 +333,7 @@ describe('update', async () => {
     });
 
     apiSenderSendMock.mockReset();
+    console.log('DA QUI 2');
```
Comment (axel7083) [inline packages/main/src/plugin/kubernetes-context-state.spec.ts]:
```suggestion
```

File: packages/main/src/plugin/kubernetes-context-state.spec.ts (hunk 1 above)
Comment (lstocchi) [reply packages/main/src/plugin/kubernetes-context-state.spec.ts]: 😓 Thanks!!

**Review by axel7083 — APPROVED** [review]
LGTM, Just tested it following the steps you provided and works like a charm!

File: packages/main/src/plugin/kubernetes-context-state.ts [hunk 3]
```diff
@@ -359,15 +359,79 @@ export class ContextsManager {
...
+  isContextChanged(kubeconfig: KubeConfig): boolean {
+    // if the current context name in the kubeconfig is different from the latest context we saved -> true
+    if (kubeconfig.currentContext !== this.currentContext?.name) {
+      return true;
+    }
+
+    // if the current context name in the kubeconfig is undefined ->  true
+    if (!kubeconfig.currentContext) {
+      return true;
+    }
```
Comment (feloy) [inline packages/main/src/plugin/kubernetes-context-state.ts]: What happens if there was already no currentContext in kubeConfig before?
It seems to me that we will detect a change every time

Comment (lstocchi) [reply packages/main/src/plugin/kubernetes-context-state.ts]: You're right, removed. Thanks!!

**Review by feloy — APPROVED** [review]

**Pull request merged**